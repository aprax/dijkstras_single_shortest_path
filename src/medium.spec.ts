import findShortestPath from '.'
import type { Nodes } from "./types";

const nodes: Nodes = new Map([
	['A', [['B', 2], ['C', 3]]],
	['B', [['A', 1], ['C', 7], ['D', 8]]],
	['C', [['A', 10], ['B', 6], ['D', 5]]],
])
nodes.forEach((_, key) => {
	it(`should find the shortest paths to all other nodes from ${key}`, () => {
		expect(findShortestPath(key, nodes)).toMatchSnapshot()
	})
});

