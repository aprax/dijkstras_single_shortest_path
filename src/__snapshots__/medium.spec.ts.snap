// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find the shortest paths to all other nodes from A 1`] = `
Map {
  "A" => Array [
    "A",
    0,
  ],
  "B" => Array [
    "A",
    2,
  ],
  "C" => Array [
    "A",
    3,
  ],
  "D" => Array [
    "C",
    8,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from B 1`] = `
Map {
  "B" => Array [
    "B",
    0,
  ],
  "A" => Array [
    "B",
    1,
  ],
  "C" => Array [
    "B",
    7,
  ],
  "D" => Array [
    "B",
    8,
  ],
}
`;

exports[`should find the shortest paths to all other nodes from C 1`] = `
Map {
  "C" => Array [
    "C",
    0,
  ],
  "D" => Array [
    "C",
    5,
  ],
  "B" => Array [
    "C",
    6,
  ],
  "A" => Array [
    "C",
    10,
  ],
}
`;
